core/src/main/scala/org/apache/spark/scheduler/TaskSchedulerImpl.scala

```diff
@@ -286,6 +286,30 @@ private[spark] class TaskSchedulerImpl(
     }
   }
 
+  /**
+   * SPARK-25250: Whenever any Result Task gets successfully completed, we simply mark the
+   * corresponding partition id as completed in all attempts for that particular stage. As a
+   * result, we do not see any Killed tasks due to TaskCommitDenied Exceptions showing up
+   * in the UI.
+   */
+  override def markPartitionIdAsCompletedAndKillCorrespondingTaskAttempts(
+      partitionId: Int, stageId: Int): Unit = {
+    taskSetsByStageIdAndAttempt.getOrElse(stageId, Map()).values.foreach { tsm =>
+      tsm.partitionToIndex.get(partitionId) match {
+        case Some(index) =>
+          tsm.markPartitionIdAsCompletedForTaskAttempt(index)
```

Comment:
shall we call `markPartitionCompleted` instead of creating this new method?

Reply:
I had this in mind when I was writing the code but a question here popped into my head, i.e.  should we really update `tasksSuccessful` counter if we are killing those tasks? Wouldn't that be incorrect? Let me know your thoughts @cloud-fan @Ngone51 .

Reply:
I think if `tsm` is the newest `TaskSet`, so, update `taskSuccessful` is necessary, so that it can have a chance to finish itself at this time. If not, I think it is not necessary, and even `successful(index) = true` is not. Because we've already send killing signals to those tasks, so zombie TaskSets will handle the failed task attempts at the end and finished normally. 

 (But you tell me `tsm` may not happen to be the newtest TaskSet above,  so I'm not sure about it. )

Reply:
sounds reasonable to me, but we should pick a better name to highlight this difference.

Reply:
Have renamed the method, please let me know if it still sounds wrong. Thank you.